samples/boards/nordic/system_off/README.rst

```diff
@@ -13,10 +13,10 @@ deep sleep on Nordic platforms.
 RAM Retention
 =============
 
-This sample can also can demonstrate RAM retention. By selecting
-``CONFIG_APP_USE_NRF_RETENTION=y`` or ``CONFIG_APP_USE_RETAINED_MEM=y``
-state related to number of boots, number of times system off was entered,
-and total uptime since initial power-on are retained in a checksummed data structure.
+This sample can also demonstrate RAM retention.
+By selecting ``CONFIG_APP_USE_RETAINED_MEM=y`` state related to number of boots,
+number of times system off was entered, and total uptime since initial power-on
+are retained in a checksummed data structure.
```

Comment:
All this code here https://github.com/zephyrproject-rtos/zephyr/pull/82262/files#diff-fca85c989d1e5dd6c706fe7388d14f46081d9de52b0603f0590cb458c3fa6342R38
Is all automatically handled simply by setting up a retained memory area then partitioning it (i.e. 100% of the area) with a retention part and enable checksum, you can see a setup and description here: https://docs.zephyrproject.org/latest/services/retention/index.html#devicetree-setup

Reply:
ah, so we would need to start using `retention` subsystem rather than `retained_mem` driver to achieve this optimization.

Would it be possible to use `retained_mem` driver API for now in this sample, and switch to `retention` service in a separate, follow-up PR ?

Reply:
Yes, switching to retention is unrelated to this PR it's just in this PR where I noticed it's reimplementing what is already available